store/tikv/split_region.go

```diff
@@ -42,32 +48,100 @@ func (s *tikvStore) SplitRegion(splitKey kv.Key) error {
 	for {
 		loc, err := s.regionCache.LocateKey(bo, splitKey)
 		if err != nil {
-			return errors.Trace(err)
+			return nil, errors.Trace(err)
 		}
 		if bytes.Equal(splitKey, loc.StartKey) {
 			logutil.Logger(context.Background()).Info("skip split region",
 				zap.Binary("at", splitKey))
-			return nil
+			return nil, nil
 		}
 		res, err := sender.SendReq(bo, req, loc.Region, readTimeoutShort)
 		if err != nil {
-			return errors.Trace(err)
+			return nil, errors.Trace(err)
 		}
 		regionErr, err := res.GetRegionError()
 		if err != nil {
-			return errors.Trace(err)
+			return nil, errors.Trace(err)
 		}
 		if regionErr != nil {
 			err := bo.Backoff(BoRegionMiss, errors.New(regionErr.String()))
 			if err != nil {
-				return errors.Trace(err)
+				return nil, errors.Trace(err)
 			}
 			continue
 		}
 		logutil.Logger(context.Background()).Info("split region complete",
 			zap.Binary("at", splitKey),
 			zap.Stringer("new region left", res.SplitRegion.GetLeft()),
 			zap.Stringer("new region right", res.SplitRegion.GetRight()))
-		return nil
+		return res.SplitRegion.GetLeft(), nil
+	}
+}
+
+func (s *tikvStore) scatterRegion(regionID uint64) error {
+	logutil.Logger(context.Background()).Info("start scatter region",
+		zap.Uint64("regionID", regionID))
+	bo := NewBackoffer(context.Background(), scatterRegionBackoff)
+	for {
+		err := s.pdClient.ScatterRegion(context.Background(), regionID)
+		if err != nil {
+			err = bo.Backoff(BoRegionMiss, errors.New(err.Error()))
+			if err != nil {
+				return errors.Trace(err)
+			}
+			continue
+		}
+		break
+	}
+	logutil.Logger(context.Background()).Info("scatter region complete",
+		zap.Uint64("regionID", regionID))
+	return nil
+}
+
+func (s *tikvStore) WaitScatterRegionFinish(regionID uint64) error {
+	logutil.Logger(context.Background()).Info("wait scatter region",
+		zap.Uint64("regionID", regionID))
+	bo := NewBackoffer(context.Background(), waitScatterRegionFinishBackoff)
+	logFreq := 0
+	for {
+		resp, err := s.pdClient.GetOperator(context.Background(), regionID)
+		if err == nil && resp != nil {
+			if !bytes.Equal(resp.Desc, []byte("scatter-region")) || resp.Status != pdpb.OperatorStatus_RUNNING {
+				logutil.Logger(context.Background()).Info("wait scatter region finished",
+					zap.Uint64("regionID", regionID))
+				return nil
+			}
+			if logFreq%10 == 0 {
+				logutil.Logger(context.Background()).Info("wait scatter region",
+					zap.Uint64("regionID", regionID),
+					zap.String("desc", string(resp.Desc)),
+					zap.String("status", pdpb.OperatorStatus_name[int32(resp.Status)]))
+			}
+			logFreq++
+		}
+		if err != nil {
+			err = bo.Backoff(BoRegionMiss, errors.New(err.Error()))
+		} else {
+			err = bo.Backoff(BoRegionMiss, errors.New("wait scatter region timeout"))
+		}
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+
+}
+
+func (s *tikvStore) SplitRegionAndScatter(splitKey kv.Key) (uint64, error) {
+	left, err := s.splitRegion(splitKey)
```

Comment:
If `splitRegion` failed will retry to split? such as `epoch is not matched`.

Reply:
There is a `Backoff` in `splitRegion`, so it will retry.